tests/framework/integration/cluster.go

```diff
@@ -1693,5 +1693,30 @@ func (c *Cluster) MustNewMember(t testutil.TB, resp *clientv3.MemberAddResponse)
 	}
 	m.InitialPeerURLsMap[m.Name] = types.MustNewURLs(resp.Member.PeerURLs)
 	c.Members = append(c.Members, m)
+
+	// Override PeerURLs and PeerListeners set by mustNewMember
+	urls := types.MustNewURLs(resp.Member.PeerURLs)
+	m.PeerURLs = urls
+
+	var listeners []net.Listener
+	for _, url := range urls {
+		var l net.Listener
+		var err error
+		switch url.Scheme {
+		case "http", "https":
+			l, err = net.Listen("tcp", url.Host)
+		case "unix", "unixs":
+			l, err = net.Listen("unix", url.Host)
+		default:
+			err = fmt.Errorf("unsupported scheme: %s", url.Scheme)
+		}
+
+		if err != nil {
+			t.Fatal("failed to listen on %v: %v", url, err)
+		}
+		listeners = append(listeners, l)
+	}
+	m.PeerListeners = listeners
```

Comment:
The test case added a learner member with peer URL `http://127.0.0.1:1234` to the cluster, but the learner member didn't listen to the host and couldn’t receive a snapshot from the leader.

Reply:
Please move to a separate PR so we can merge this faster.